pkg/codegen/go/gen.go

```diff
@@ -879,20 +912,56 @@ func (pkg *pkgContext) genConfig(w io.Writer, variables []*schema.Property) erro
 	return nil
 }
 
+// GoInfo holds information required to generate the Go SDK from a schema.
+type GoInfo struct {
+	// Map from module -> package name
+	//
+	//     { "flowcontrol.apiserver.k8s.io/v1alpha1": "flowcontrol/v1alpha1" }
+	//
+	ModuleToPackage map[string]string `json:"moduleToPackage,omitempty"`
+
+	// Map from package name -> package alias
+	//
+	//     { "github.com/pulumi/pulumi-kubernetes/sdk/go/kubernetes/flowcontrol/v1alpha1": "flowcontrolv1alpha1" }
+	//
+	PackageImportAliases map[string]string `json:"packageImportAliases,omitempty"`
+}
+
 func GeneratePackage(tool string, pkg *schema.Package) (map[string][]byte, error) {
+	var goInfo GoInfo
+	err := json.Unmarshal(pkg.Language["go"], &goInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	// importBasePath converts from a repository URL to an import path.
+	//   https://github.com/pulumi/pulumi-kubernetes -> github.com/pulumi/pulumi-kubernetes/sdk/go/kubernetes
+	importBasePath := func() string {
```

Comment:
It might make more sense to put this in the GoInfo rather than computing it.

Reply:
Yes--that was what I was suggesting :)